kv/txn_coord_sender.go

```diff
@@ -54,22 +55,27 @@ import (
 // periodically on their own.
 type txnMetadata struct {
 	// txn is the transaction struct from the initial AddRequest call.
-	txn proto.Transaction
```

Comment:
Why make a copy of the transaction? When we init this txnMetadata, the transaction we assign here is already heap-allocated. Might as well just use that.